app/src/lib/stores/app-store.ts

```diff
@@ -4617,24 +4638,39 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this.emitUpdate()
   }
 
-  public _setDivergingBranchBannerVisibility(
+  /** This shouldn't be called directly. See `Dispatcher`. */
+  public _updateDivergingBranchBannerState(
```

Comment:
This method now can update any of the 3 NBBD flags by passing an object with the flags to update. This reduces some repetition and makes sure that only 1 update is emitted when changing multiple flags at once.

The main tradeoff is that the method is a bit more "dynamic" in the sense that it accepts a `Partial` of an interface as an argument (still, everything is still correctly covered by TypeScript).